create_sarek_samplesheet.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import re
+
+fqdir = sys.argv[1]
+outfile = sys.argv[2]
+
+sampledict = {}
+fqpattern = r'^(.*)_(S\d+)_L(\d+)_(R\d)_001.*$'
+fcpattern = r'^.*_[AB]?([^_]+)$'
+
+
+for root, dirs, fqlinks in os.walk(fqdir):
+    for link in fqlinks:
+        fqpath = os.readlink(os.path.join(root, link))
+        fqbasenm = re.match(fqpattern, link)
+
+        if fqbasenm:
+            sample = fqbasenm.group(1)
+            ssheet_idx = fqbasenm.group(2)
+            laneno = int(fqbasenm.group(3))
+            readnr = fqbasenm.group(4)
+
+        else:
+            continue
+
+        m = re.match(fcpattern, os.path.basename(os.path.dirname(fqpath)))
+        fcid = m.group(1) if m else "NA"
+
+        readgrp = f"{fcid}.{laneno}.{ssheet_idx}"  # PU, Plattform unit. Will be used as read tag in BAM-files.
+
+        if readgrp not in sampledict:
+            sampledict[readgrp] = {"sample": sample, "R1": "", "R2": ""}
+        sampledict[readgrp][readnr] = fqpath
+
+with open(outfile, 'w') as fout:
+    fout.write("patient,sample,lane,fastq_1,fastq_2\n")
+    for readgrp in sampledict:
+        samplenm = sampledict[readgrp]["sample"]
+        R1 = sampledict[readgrp]["R1"]
+        R2 = sampledict[readgrp]["R2"]
+        entry = ",".join([samplenm, samplenm, readgrp, R1, R2])
```

Comment:
Just double-checking: Should readgrp be set as lane?

Reply:
yes, it's just the header that is a bit strange for the sarek samplesheets.